### ISSUE TITLE
'call' and '!' not behaving the same

### ISSUE THREAD

**Issue by tankorsmash** [uiua-lang/uiua#53]
This [pad](https://www.uiua.org/pad?src=YXJyIOKGkCBb4pahICIxIiDilqEgIjIiXQoK4oi1Y2FsbCBhcnI=)
```
arr ← [□ "1" □ "2"]

∵call arr
```

throws the following error:
```
Error: Stack was empty when evaluating index
   ╭─[:3:2]
 3 │ ∵call arr
   │  ────  
───╯
  in ! at 3:2
```

but the following [pad](https://www.uiua.org/pad?src=YXJyIOKGkCBb4pahICIxIiDilqEgIjIiXQoK4oi1ISBhcnI=) does not

```
arr ← [□ "1" □ "2"]

∵! arr
```

**Comment by tankorsmash** [comment]
Discussing in [the Discord](https://discord.com/channels/1156339038748413952/1156817423048118343/1158154905958232114), there seems to be an issue with how Uiua handles long names at the moment:

> Kaikalii: The interpreter isn't supposed to interpret the names of functions that have ascii glyphs i.e. !+-'~ etc
> Kaikalii: `call` as you used it should have been an error
> [...]
> Kaikalii: What's even worse is it doesn't format it.

**Comment by kaikalii** [comment]
Okay, so the parser saw "call" and said "I know that's the name of a built-in function, but you can just type it as '!', so I'll parse that as an identifier.
The formatter only acts on output from the parser, so it said "Yep, just an identifer. Print it can carry on".
But then the *compiler* said, "Wait, but there's a function called 'call'. That's definitely what they're talking about."

Fixed in 3b34477c888af85
You will now get errors like this:
```
Error: Unknown identifier `call`, did you mean `!`?
   ╭─[main.ua:2:2]
 2 │ ∵call arr
   │  ────
───╯
```

**Comment by tankorsmash** [comment]
Thank you!
